OpenRA.Mods.Cnc/Activities/LayMines.cs

```diff
@@ -141,14 +145,11 @@ static bool CanLayMine(Actor self, CPos p)
 
 		void LayMine(Actor self)
 		{
-			if (ammoPools != null)
-			{
-				var pool = ammoPools.FirstOrDefault(x => x.Info.Name == minelayer.Info.AmmoPoolName);
-				if (pool == null)
-					return;
+			var pool = ammoPools.FirstOrDefault();
```

Comment:
This also needs to be
```suggestion
			var pool = ammoPools.FirstOrDefault(p => p.CurrentAmmoCount >= minelayer.Info.AmmoUsage);
```
now.